### PR TITLE
Feature/chat message multi server

### PR DESCRIPTION
**Pull request by thanhptvmodev**
Chat message multiserver - Expt: Share photo, call message

### CODE REVIEW

File: ClearKeep/ClearKeep/Classes/Screens/Home/CreateGroup/InviteMemberGroup.swift [hunk 1]
```diff
@@ -130,4 +178,32 @@ extension InviteMemberGroup {
             }
         }
     }
+    
+    func getUserInfo() {
+        //Ex: 54.235.68.160:25000:69b14823-9612-4fa4-9023-f11351e921e2
+        let url = "54.235.68.160:25000:69b14823-9612-4fa4-9023-f11351e921e2".trimmingCharacters(in: .whitespacesAndNewlines)
```
Comment (namnh92) [inline ClearKeep/ClearKeep/Classes/Screens/Home/CreateGroup/InviteMemberGroup.swift]: @thanhptvmodev : Please explain about this URL

File: ClearKeep/ClearKeep/Classes/Screens/Home/CreateGroup/InviteMemberGroup.swift [hunk 2]
```diff
@@ -130,4 +178,32 @@ extension InviteMemberGroup {
...
+        guard let workspaceDomain = url.components(separatedBy: ":").first,
+              let userId = url.components(separatedBy: ":").last else {
+            return
+        }
```
Comment (namnh92) [inline ClearKeep/ClearKeep/Classes/Screens/Home/CreateGroup/InviteMemberGroup.swift]: @thanhptvmodev : Create a function return tuple (domain, userId)  

```suggestion
        func getInfo(from url: String) -> (String, String)
```

File: ClearKeep/ClearKeep/Classes/Networking/Backend.swift [hunk 3]
```diff
@@ -280,6 +279,23 @@ class Backend {
         }
     }
     
+    func getUserInfo(userId: String, workspaceDomain: String, _ completion: @escaping (User_UserInfoResponse?, Error?) -> Void) {
```
Comment (namnh92) [inline ClearKeep/ClearKeep/Classes/Networking/Backend.swift]: @thanhptvmodev : 
```suggestion
    func getUserInfo(userId: String, workspaceDomain: String, _ completion: @escaping (Result<User_UserInfoResponse?, Error?>) -> Void) {
```

File: ClearKeep/ClearKeep/Classes/Screens/AlbumsView/AlbumsView.swift [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+//
+//  Created by Diflega on 9/6/21.
+//
+
+import Foundation
+import SwiftUI
+
+struct AlbumsView: View {
+    
+    //@Environment(\.presentationMode) var presentationMode: Binding<PresentationMode>
```
Comment (namnh92) [inline ClearKeep/ClearKeep/Classes/Screens/AlbumsView/AlbumsView.swift]: @thanhptvmodev Remove comment

File: ClearKeep/ClearKeep/Classes/Screens/AlbumsView/AlbumsView.swift [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        }
+        .applyNavigationBarPlainStyleLight(title: "", leftBarItems: {
+            Button {
+                //presentationMode.wrappedValue.dismiss()
+                dismissAlert = false
+            } label: {
+                Image("ic_close")
+                    .frame(width: 24, height: 24)
+                    .foregroundColor(AppTheme.colors.offWhite.color)
+            }
```
Comment (namnh92) [inline ClearKeep/ClearKeep/Classes/Screens/AlbumsView/AlbumsView.swift:23]: @thanhptvmodev : Create common button Close and Back

File: ClearKeep/ClearKeep/Classes/Screens/Authentication/AdvanceServerSettings/AdvanceServerSettingsView.swift [hunk 6]
```diff
@@ -78,17 +78,20 @@ struct AdvanceServerSettingsView: View {
     }
     
     private func useCustomServer() {
-        guard let first = customServerURL.components(separatedBy: ":").first,
-              let last = customServerURL.components(separatedBy: ":").last else {
+        let url = customServerURL.trimmingCharacters(in: .whitespacesAndNewlines)
+        guard let first = url.components(separatedBy: ":").first,
+              let last = url.components(separatedBy: ":").last else {
```
Comment (namnh92) [inline ClearKeep/ClearKeep/Classes/Screens/Authentication/AdvanceServerSettings/AdvanceServerSettingsView.swift]: @thanhptvmodev : Create function return tuple, and explain first & last for more meaning

File: ClearKeep/ClearKeep/Classes/Screens/Authentication/Login/LoginViewModel.swift [hunk 7]
```diff
@@ -17,7 +17,17 @@ struct UserLoginResponseInfo {
...
     @Published var isUseCustomServer = false
     @Published var customServerURL = ""
-    @Published var customServerPort = ""
     
     var userLoginResponseInfo = UserLoginResponseInfo(userId: "", userDisplayName: "", userEmail: "", signInType: .email)
+    
+    func usedCustomServer() -> Bool {
+        guard let first = customServerURL.components(separatedBy: ":").first,
+              let last = customServerURL.components(separatedBy: ":").last else {
+            return false
```
Comment (namnh92) [inline ClearKeep/ClearKeep/Classes/Screens/Authentication/Login/LoginViewModel.swift]: @thanhptvmodev : Create function return tuple, and explain first & last for more meaning

File: ClearKeep/ClearKeep/Classes/Screens/Home/HomeMain/HomeMainViewModel.swift [hunk 8]
```diff
@@ -39,4 +60,9 @@ class HomeMainViewModel: ObservableObject {
             })
         }
     }
+    
+    func getUsers() -> [User] {
```
Comment (namnh92) [inline ClearKeep/ClearKeep/Classes/Screens/Home/HomeMain/HomeMainViewModel.swift]: @thanhptvmodev : Make private access

**Comment by thanhptvmodev** [comment]
Create other branch

**Pull request closed without merging**